src/host/snapshot.h

```diff
@@ -13,13 +13,42 @@ namespace fs = std::filesystem;
 
 namespace asynchost
 {
+  static constexpr auto snapshot_file_prefix = "snapshot";
+  static constexpr auto snapshot_idx_delimiter = "_";
+  static constexpr auto snapshot_committed_suffix = "committed";
+
+  size_t get_snapshot_idx_from_file_name(const std::string& file_name)
+  {
+    auto pos = file_name.find(snapshot_idx_delimiter);
+    if (pos == std::string::npos)
+    {
+      throw std::logic_error(fmt::format(
+        "Snapshot file name \"{}\" does not contain seqno", file_name));
+    }
+
+    return std::stol(file_name.substr(pos + 1));
+  }
+
+  std::optional<size_t> get_snapshot_evidence_idx_from_file_name(
+    const std::string& file_name)
+  {
+    auto pos = file_name.find(
+      fmt::format("{}{}", snapshot_committed_suffix, snapshot_idx_delimiter));
+    if (pos == std::string::npos)
+    {
+      // Snapshot is not yet committed
+      return std::nullopt;
+    }
+
+    auto committed_suffix = file_name.substr(pos);
+    return std::stol(committed_suffix.substr(
```

Comment:
This is returning a long as a size_t, so this would technically accept a negative value which would come out as a large unsigned. `std::from_chars()` is more precise and would validate this away.

This is untrusted code anyway, but since it's not much more verbose, I think it's still reasonable to make the change.